metricbeat/module/kubernetes/kubernetes.go

```diff
@@ -39,6 +40,7 @@ func init() {
 type Module interface {
 	mb.Module
 	GetSharedFamilies(prometheus p.Prometheus) ([]*dto.MetricFamily, error)
+	GetSharedKubeletStats(http *helper.HTTP) ([]byte, error)
```

Comment:
Nit. I am thinking now that this being shared is an implementation detail, interface and consumers don't care if the response is shared or cached.
```suggestion
	GetFamilies(prometheus p.Prometheus) ([]*dto.MetricFamily, error)
	GetKubeletStats(http *helper.HTTP) ([]byte, error)
```